code/modules/discord/discord_sql_functions.dm

```diff
@@ -0,0 +1,198 @@
+
+/**
+ * Generate a timebound token for discord verification
+ *
+ * This uses the common word list to generate a six word random token, this token can then be fed to a discord bot that has access
+ * to the same database, and it can use it to link a ckey to a discord id, with minimal user effort
+ *
+ * It returns the token to the calling proc, after inserting an entry into the discord_link table of the following form
+ *
+ * ```
+ * (unique_id, ckey, null, the current time, the one time token generated)
+ * the null value will be filled out with the discord id by the integrated discord bot when a user verifies
+ * ```
+ *
+ * Notes:
+ * * The token is guaranteed to be unique during it's validity period
+ * * The validity period is currently set at 4 hours
+ * * a token may not be unique outside it's validity window (to reduce conflicts)
+ *
+ * Arguments:
+ * * ckey_for a string representing the ckey this token is for
+ *
+ * Returns a string representing the one time token
+ */
+/client/proc/discord_generate_one_time_token(ckey_for)
+	var/static/list/token_wordlist = world.file2list("strings/eff_large_wordlist.txt")
+	var/not_unique = TRUE
+	var/one_time_token = ""
+	// While there's a collision in the token, generate a new one (should rarely happen)
+	while(not_unique)
+		//Column is varchar 100, so we trim just in case someone does us the dirty later
+		one_time_token = trim("[pick(token_wordlist)]-[pick(token_wordlist)]-[pick(token_wordlist)]", 100)
+
+		not_unique = find_discord_link_by_token(one_time_token, timebound = TRUE)
+
+	// Insert into the table, null in the discord id, id and timestamp and valid fields so the db fills them out where needed
+	var/datum/db_query/query_insert_link_record = SSdbcore.NewQuery({"
+		INSERT INTO [format_table_name("discord_links")] (
+			ckey,
+			one_time_token
+		) VALUES (
+			:ckey,
+			:token
+		)"},
+		list(
+			"ckey" = ckey_for,
+			"token" = one_time_token
+		)
+	)
+
+	if(!query_insert_link_record.Execute())
+		qdel(query_insert_link_record)
+		return ""
```

Comment:
This should probably have some sort of error logging or failsafe added to it. Currently it just continues on as normal with an empty string as the OTP.

Reply:
```suggestion
	if(!query_insert_link_record.Execute())
		qdel(query_insert_link_record)
		message_admins("WARNING! FAILED TO ASSIGN A OTP FOR USER [ckey_for], CALL A CODER!")]
		stack_trace("WARNING! FAILED TO ASSIGN A OTP FOR USER [ckey_for], CALL A CODER!")
		return "INVALID_PASSWORD_CALL_A_CODER"
```